lib/main_common.pm

```diff
@@ -2489,6 +2489,26 @@ sub load_security_tests_tpm2 {
     }
 }
 
+sub load_security_tests_create_swtpm_hdd {
+    load_security_console_prepare;
+
+    loadtest "security/create_swtpm_hdd/build_hdd";
+}
+
+sub load_security_tests_swtpm {
+    load_security_console_prepare;
+
+    loadtest "security/swtpm/swtpm_env_setup";
+    loadtest "security/swtpm/swtpm_verify";
+}
+
+sub load_security_tests_swtpm_uefi {
```

Comment:
This appears to be identical to the function without `_uefi` above. Whether it uses uefi or not depends entirely on `HDD_SWTPM_UEFI`, so this can probably be removed.

Reply:
Good comments, the original logic was the same as you mentioned above. 

However, for my tests, I need cover both Legacy and UEFI vm. then, I should create 2 predefined qcow2 files.
(one for Legacy, and the other for UEFI). and both of them should run upon my L1 vm (Legacy mode).

In this case, I have 2 dependency trees for final "swtpm_verify" case.

1. create-hdd-swtpm->swtpm_verify
2. create_hdd_uefi->create-hdd-swtpm-uefi->swtpm_verify

It seems make my test more complex since I get so many dependencies there.  as discussed within my team, I decided to spit it into 2 individual cases.

Reply:
Neither of those dependency trees need `load_security_tests_swtpm_uefi`.

Reply:
I defined parameters in test suite, since both Lagacy and UEFI L2 vm guests need build upon the L1 VM (legacy mode).

**create_swtpm_hdd_uefi:**
BOOT_HDD_IMAGE=1
BOOT_MENU=1
DESKTOP=textmode
DISTRI=sle
FLAVOR=Online
HDD_1=%DISTRI%-%VERSION%-%ARCH%-%BUILD%-%DESKTOP%@%MACHINE%.qcow2
MACHINE=uefi
PUBLISH_HDD_1=%DISTRI%-%VERSION%-%ARCH%-%BUILD%-%DESKTOP%@swtpm-uefi.qcow2
SECURITY_TEST=create_swtpm_hdd
START_AFTER_TEST=create_hdd_textmode_uefi
UEFI=1
VIDEOMODE=text
---------
**create_swtpm_hdd:**
BOOT_HDD_IMAGE=1
DESKTOP=textmode
DISTRI=sle
FLAVOR=Online
HDD_1=%DISTRI%-%VERSION%-%ARCH%-%BUILD%-%DESKTOP%@%MACHINE%.qcow2
MACHINE=64bit
PUBLISH_HDD_1=%DISTRI%-%VERSION%-%ARCH%-%BUILD%-%DESKTOP%@swtpm-legacy.qcow2
SECURITY_TEST=create_swtpm_hdd
START_AFTER_TEST=create_hdd_textmode
VIDEOMODE=text

--------
**security_swtpm_uefi**:
BOOT_HDD_IMAGE=1
DESKTOP=textmode
DISTRI=sle
FLAVOR=Online
HDD_1=%DISTRI%-%VERSION%-%ARCH%-%BUILD%-%DESKTOP%@%MACHINE%.qcow2
HDD_SWTPM_UEFI=%DISTRI%-%VERSION%-%ARCH%-%BUILD%-%DESKTOP%@swtpm-uefi.qcow2
MACHINE=64bit
QEMUCPU=host
QEMURAM=4096
QEMU_MAX_MIGRATION_TIME=500
SECURITY_TEST=swtpm_uefi
START_AFTER_TEST=create_swtpm_hdd_uefi@uefi
VIDEOMODE=text

----------
**security_swtpm:**
BOOT_HDD_IMAGE=1
DESKTOP=textmode
DISTRI=sle
FLAVOR=Online
HDD_1=%DISTRI%-%VERSION%-%ARCH%-%BUILD%-%DESKTOP%@%MACHINE%.qcow2
HDD_SWTPM_LEGACY=%DISTRI%-%VERSION%-%ARCH%-%BUILD%-%DESKTOP%@swtpm-legacy.qcow2
MACHINE=64bit
QEMUCPU=host
QEMURAM=4096
QEMU_MAX_MIGRATION_TIME=500
SECURITY_TEST=swtpm
START_AFTER_TEST=create_swtpm_hdd
VIDEOMODE=text

Reply:
What I mean is that you can just replace `SECURITY_TEST=swtpm_uefi` with `SECURITY_TEST=swtpm` and drop the function without any difference.

Reply:
Got it, thanks, let me fix it.